### PR TITLE
GAWB-956: added delete google billing admin endpoint

### PR DESCRIPTION
**Pull request by skwalker**
- [ ] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: Make sure documentation for code is complete
- [ ] **Submitter**: Make sure liquibase is updated if appropriate. 
  - If doing a migration, take a backup of the dev and alpha DBs in Google Cloud Console
- [ ] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [ ] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [ ] **Submitter**: If PR includes new or changed db queries, include the explain plans in the description
- [ ] **Submitter**: Update FISMA documentation if changes to:
  - Authentication
  - Authorization
  - Encryption
  - Audit trails
- [ ] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] Tell  that the PR exists if he wants to look at it **(apply requires_doge label)**
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- Review cycle:
  - LR reviews
  - Rest of team may comment on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH
  - Submitter updates documentation as needed
  - Submitter **reassigns to LR** for further feedback
- [ ]  sign off
- [x] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Squash commits, rebase if necessary
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiService.scala [hunk 1]
```diff
@@ -37,15 +37,24 @@ trait AdminApiService extends HttpService with PerRequestCreator with UserInfoDi
       }
     } ~
     path("admin" / "billing" / Segment) { (projectId) =>
+      delete {
+        requestContext => perRequest(requestContext,
+          UserService.props(userServiceConstructor, userInfo),
+          UserService.AdminDeleteBillingProject(RawlsBillingProjectName(projectId)))
+      }
+    } ~
+    path("admin" / "billing" / "register" / Segment) { (projectId) =>
```
Comment (skwalker) [inline src/main/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiService.scala:46]: is this naming ok?

Comment (ansingh7115) [reply src/main/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiService.scala:46]: It looks good to me

**Comment by ansingh7115** [comment]
👍

File: src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala [hunk 2]
```diff
@@ -648,6 +648,22 @@ class HttpGoogleServicesDAO(
     }
   }
 
+  override def deleteProject(projectName: RawlsBillingProjectName): Future[Unit]= {
+    val resMgr = getCloudResourceManager(getBillingServiceAccountCredential)
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleServicesDAO.scala]: call getBillingServiceAccountCredential once and reuse it

File: src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala [hunk 3]
```diff
@@ -353,7 +355,13 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     }
   }
 
-  def createBillingProject(projectName: RawlsBillingProjectName): Future[PerRequestMessage] = dataSource.inTransaction { dataAccess =>
+  def deleteBillingProject(projectName: RawlsBillingProjectName): Future[PerRequestMessage] = {
+    // delete actual project in google-y way, then remove from Rawls DB
+    DBIO.from(gcsDAO.deleteProject(projectName))
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala]: the result of this future is lost. This function should probably look like
gcsDAO.deleteProject(projectName) flatMap { _ => unregisterBillingProject(projectName) }

**Comment by dvoet** [comment]
👍

**Pull request merged**